projects/packages/post-list/src/class-post-list.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * The PostList Admin Area.
+ *
+ * @package automattic/jetpack-post-list
+ */
+
+namespace Automattic\Jetpack\PostList;
+
+use Automattic\Jetpack\Assets;
+
+require_once __DIR__ . '/class-post-thumbnail.php';
+
+/**
+ * The PostList Admin Area
+ */
+class Post_List {
+	const PACKAGE_VERSION = '1.0.0-alpha';
+
+	/**
+	 * The configuration method that is called from the jetpack-config package.
+	 */
+	public static function configure() {
+		$post_list = self::get_instance();
+		$post_list->register();
+	}
+
+	/**
+	 * Check whether the wp-admin-posts-list-page feature is enabled,
+	 * via the query string.
+	 *
+	 * @return boolean True when feature is active. Otherwise, False.
+	 */
+	public static function is_wp_admin_posts_list_page() {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		return isset( $_GET['post-list'] ) && 'true' === $_GET['post-list'];
+	}
+
+	/**
+	 * Initialize the Post List UI.
+	 *
+	 * @return Post_List Post_List instance.
+	 */
+	public static function get_instance() {
+		if ( self::is_wp_admin_posts_list_page() ) {
+			return new Post_List();
+		}
+	}
+
+	/**
+	 * Sets up Post List action callbacks if needed.
+	 */
+	public function register() {
+		if ( ! did_action( 'jetpack_on_posts_list_init' ) ) {
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+
+			/**
+			 * Action called after initializing PostList Admin resources.
+			 *
+			 * @since $$next-version$$
+			 */
+			do_action( 'jetpack_on_posts_list_init' );
+		}
+	}
+
+	/**
+	 * Enqueue scripts depending on the post-list query var.
+	 *
+	 * @param string $hook Page hook.
+	 */
+	public function enqueue_scripts( $hook ) {
+		if ( 'edit.php' === $hook ) {
+			$build_assets = require_once __DIR__ . '/../build/index.asset.php';
+			$plugin_path  = Assets::get_file_url_for_environment( '../build/index.js', '../build/index.js', __FILE__ );
+
+			wp_enqueue_script(
+				'jetpack_posts_list_ui_script',
```

Comment:
I've added a call to `wp_set_script_translations` using `jetpack` as the domain. It made me think about language packs for the package. If it's used on a site without the Jetpack plugin, would the translations be available?

Reply:
> If it's used on a site without the Jetpack plugin, would the translations be available?

No, because the textdomains would have to be different. We don't really have a nice way around that problem right now. This is tracked in #20320